airbyte-webapp/src/components/connection/ConnectionForm/EditControls.tsx

```diff
@@ -25,24 +29,23 @@ const EditControls: React.FC<EditControlProps> = ({
   successMessage,
   errorMessage,
   enableControls,
+  hidden,
 }) => {
+  const isButtonDisabled = hidden || isSubmitting || (!dirty && !enableControls);
   return (
-    <FlexContainer justifyContent="flex-end" alignItems="center" direction="row" gap="lg" className={styles.container}>
+    <FlexContainer
+      justifyContent="flex-end"
+      alignItems="center"
+      direction="row"
+      gap="lg"
+      className={classNames(styles.container, { [styles.cloud]: isCloudApp(), [styles.hidden]: hidden })}
```

Comment:
I feel moving those buttons just when the intercom button is present feels a bit weird, and like a pattern that will cause us eventually more work trying to maintain. Are we taking the intercom button at all into consideration when making designs @Upmitt?

Would it make sense in OSS just to accept that the form buttons are not "fully right aligned" and instead keep the space for where we expect in cloud to have the intercom button?